internal/provisioner/nginx.go

```diff
@@ -116,7 +116,11 @@ func (n *nginx) ActualVersion() *model.HelmUtilityVersion {
 }
 
 func (n *nginx) Destroy() error {
-	return nil
+	helm, err := n.NewHelmDeployment()
+	if err != nil {
+		return err
+	}
+	return helm.Delete()
```

Comment:
I believe the helm client would return immediately after calling delete, but k8s may need some time to clean things up in the background afterwards. Do you know if that is correct or not? I am wondering if we need to add some wait logic to ensure there is enough time for a full cleanup.